sdk/servicebus/service-bus/src/serviceBusMessage.ts

```diff
@@ -1040,7 +1053,13 @@ export class ServiceBusMessageImpl implements ReceivedMessageWithLock {
     }
     const receiver = this.getReceiverFromContext();
     this.throwIfMessageCannotBeSettled(receiver, DispositionType.deadletter);
-
+    if (!receiver || !receiver.isOpen()) {
+      await this.backUpMessageSettlementThroughMgmtLink(
```

Comment:
Seems like this function supercedes the old way of just calling updateDispositionStatus? So can we make the old block (where we check the map) and this one?

Reply:
Did something similar + some refactoring, please review again.